src/components/common/PinnedItems.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types'
+import { makeStyles, useTheme } from '@material-ui/core/styles';
+import { Box, Paper, Slide } from '@material-ui/core';
+import IconButton from '@material-ui/core/IconButton';
+import VisibilityIcon from '@material-ui/icons/Visibility';
+import VisibilityOffIcon from '@material-ui/icons/VisibilityOff'
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import { HEADER_HEIGHT } from '../../config/constants';
+
+const drawerWidth = 400;
+
+const useStyles = makeStyles((theme) => ({
+  iconButton:{
+    float: 'right'
+  },
+  root: {
+    display: 'flex',
+  },
+  drawer: {
+    width: drawerWidth,
+    flexShrink: 0,
+  },
+  drawerHeader: {
+    display: 'flex',
+    alignItems: 'center',
+    padding: theme.spacing(0, 1),
+    justifyContent: 'flex-start',
+  },
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing(3),
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+  },
+  contentShift: {
+    transition: theme.transitions.create('margin', {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginRight: 0,
+  },
+}));
+
+export default function PersistentDrawerRight({ children, sideContent }) {
```

Comment:
`PinnedItems`

Reply:
But actually this component is rendering the whole content (and not just the pinned items). I would split this component into two components: The main content (you can call it `Main` if there isn't one already) and the actual PinnedItems``

Reply:
This has been changed in https://github.com/graasp/graasp-perform/pull/31